### PR TITLE
simplecv depth/mesh helpers; PromptDA teacher batch tool GPU cast + malloc_trim (stack 2/7)

### PR DESCRIPTION
**Pull request by pablo-rerun**
Stack **2/9** — the ZipDepth-PromptDA work re-cut into one focused stack (replaces #132 #133 #134 #129 #164 #165 #166 #167 #172 #173 #174 #175 #177; #131 is already merged). Review bottom-up.

| # | PR | what | lines (no lock) |
|---|---|---|---:|
| 1 | #186 | workspace rerun-sdk 0.37.0 (pins + API ports) — lock | +533/−317 |
| 2 | #187 | simplecv depth/mesh helpers; PromptDA teacher batch tool on the GPU + malloc_trim; catalog-server rules | +196/−74 |
| 3 | #188 | monoprior config-first depth-completion zoo (PromptDA torch/onnx/trt) — refactor only | +752/−565 |
| 4 | #189 | **ZipDepth-PromptDA model** + static-batch TensorRT export — lock (+trtkit) | +1132/−11 |
| 5 | #190 | zipdepth-catalog lane: streaming dataset, CUDA builders, holdout eval — lock (+2 envs) | +3176/−25 |
| 6 | #191 | catalog fine-tune trainer + live smoke gate + IO instrumentation | +1576/−27 |
| 7 | #192 | hard-frame mining, fixed hard-20 scorer, Polycam three-way comparison | +1727/−14 |
| 8 | #203 | training-speed knobs + named presets for the wall-clock hill-climb | +670/−76 |
| 9 | #204 | idiomatic Rerun 0.37 dataloader as executable reference (`--dataloader rerun`), catalog-query audit | +1027/−54 |

Whole stack vs main: 108 files, +8486/−995. The old chain was 13 open PRs (+10.4k lines) plus 11 unpushed commits; the head-variant experiments and the default-off loss knobs were dropped (negative results, see the [project report](https://pablos-4800gt.ilish-ruler.ts.net:8768/zipdepth/zipdepth-promptda-final-report.html)), and every slice went through a four-lens simplify pass before cutting. Review-pass changes (2026-09-03) are folded into the same PRs: zero `Any` on added lines, jaxtyping dtype+shape on every added array annotation, no dtype/shape in new identifiers, shared metre→mm quantiser, `upsample_depth_to_image`, condensed AGENTS section, stale einops suppressions dropped in touched files.

## simplecv depth/mesh helpers; PromptDA teacher batch tool on the GPU, malloc_trim per segment
**What** `simplecv.ops.depth.quantize_depth_m_to_mm` (torch, on-device) and `simplecv.ops.tsdf_depth_fuser.log_fused_mesh` (typed `mesh: o3d.geometry.TriangleMesh`, beside the fuser that produces the meshes; entity-first; keyword `recording`/`static`/`face_rendering`) replace inlined copies in view_polycam_data, monoprior polycam_inference and the prompt-da ARKitScenes tools.
The chosen-frame PromptDA batch tool (`prompt-da-arkitscenes`) casts depth to uint16 and downsizes the fusion RGB on the GPU before the D2H copy (22.4 → 6.3 MB/frame, 17.3 → 21.6 frames/s on the 5090; stored bytes and layer contract unchanged; A/B mesh vertex count identical) and returns Open3D's TSDF arena pages to the OS at each segment boundary (`gc.collect()` + `malloc_trim(0)`), which fixed the ~0.6 GB/segment host growth that OOM-killed the corpus batch at 142 GB RSS (RSS now 4–8 GB across 570 segments).
AGENTS.md gains the catalog-server rules from that incident: never kill Rerun by name, paseo's cgroup `KillMode`, tmux / systemd-run for long jobs.
**Gates** simplecv-dev 152 passed; prompt-da-dev 8 passed + deadcode; monoprior-dev lint/typecheck.

### CODE REVIEW

File: packages/simplecv/simplecv/rerun_log_utils.py [hunk 1]
```diff
@@ -507,6 +507,40 @@ def mesh_bounding_geometry(vertices: Float64[ndarray, "n 3"]) -> tuple[Float64[n
     return center, radius
 
 
+def log_fused_mesh(
+    entity_path: str,
+    mesh: Any,
```
Comment (pablo-rerun) [inline packages/simplecv/simplecv/rerun_log_utils.py]: we should be more specific here, this shouldn't just be Any. We should never use any unless theres absolutely no way around it.

File: AGENTS.md [hunk 2]
```diff
@@ -109,6 +109,41 @@ layer. `arkitscenes-download-serve` uses `ulimit -n 524288`; restart the server
 to inherit it. This is a capacity workaround—the upstream fix is open-on-demand
 files or an LRU descriptor pool.
 
+### The catalog server is in-memory and dies to `pkill -x rerun`
```
Comment (pablo-rerun) [inline AGENTS.md]: this is good to have but too verbose, can you make it more concise please?

File: packages/simplecv/simplecv/rerun_log_utils.py (hunk 1 above)
Comment (pablo-rerun) [reply packages/simplecv/simplecv/rerun_log_utils.py]: Fixed, and applied stack-wide. `log_fused_mesh` moved beside the Open3D fuser that produces these meshes (`simplecv.ops.tsdf_depth_fuser`) and is typed `mesh: o3d.geometry.TriangleMesh`; it could not stay in `rerun_log_utils` because 40 of the 66 envs that install simplecv have no open3d. I then audited every `Any` the whole stack adds (58 lines across 15 files) and removed all of them: typed pyserde configs/reports instead of `dict[str, Any]`, a `SegmentRow` dataclass for catalog rows, `runtime_checkable` Protocols for the instrumented dataset and the export model, a `TypedDict` for albumentations results, and validated `dict[str, Tensor]` narrowing of `torch.load` results. `git diff main -- '*.py' | grep '^+' | grep -w Any` is empty on every PR of the stack.

File: AGENTS.md (hunk 2 above)
Comment (pablo-rerun) [reply AGENTS.md]: Condensed from 35 to 22 lines: same two rules (never kill Rerun by name; paseo's cgroup `KillMode` means long-lived jobs go through tmux or `systemd-run --user`), the incident dates and the 142 GB anecdote dropped, the "real fixes" kept as one sentence.
